### ISSUE TITLE
Incorrect handling of remote relative $refs

### ISSUE THREAD

**Issue by peteslater-ee** [swagger-api/swagger-parser#1745]
We have specification yaml file containing paths with relative $refs to yaml files in a subdirectory.

Like so:
```
/
application.yaml
pathParameters.yaml
endpoints/
  list.yaml
```
Inside of `application.yaml` we have paths that references files in the `endpoints` path segment.

For example:
```
paths:
  /individuals/calculations/{nino}/self-assessment:
    $ref: "./endpoints/list.yaml"
```

Those yaml files contain relative $refs for parameters to yaml files in the parent path.

For example:
```
parameters:
  - $ref: '../pathParameters.yaml#/components/parameters/nino'
```

These files are served over http and not resolved on the file system or classpath.
This seems to work when reading from a file system but does not work when reading from url.

[buildUrl](https://github.com/swagger-api/swagger-parser/blob/master/modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/RefUtils.java#L152) in RefUtils is given `http://some.main.com/aaa/bbb/application.yaml` and `./endpoints/../pathParameters.yaml` and strips the first two path fragments from it but also strips one too many path fragments from the root location due to how the `..` is processed.

The actual result from the parser is `http://some.main.com/aaa/pathParameters.yaml`
The expected result is `http://some.main.com/aaa/bbb/pathParameters.yaml`

We believe this to be due to the code in [line 81](https://github.com/swagger-api/swagger-parser/blob/master/modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/RefUtils.java#L181) of RefUtils.
This code works when the `..` is at the start of the relative path. When the `..` in not at the start, as the example above shows, the code incorrectly builds the url.

When have altered a local copy of the code to change how the `buildUrl` method works which fixes this problem. This change doesn't break of the existing tests.

We have opted to remove the custom url handling in the code and instead pass the root url and relative path to `java.net.URL` which works out the actual path.

```
    public static String buildUrl(String rootPath, String relativePath) {
        if(rootPath == null || relativePath == null) {
            return null;
        }

        try {
            int until = rootPath.lastIndexOf("/")+1;
            String root = rootPath.substring(0, until);
            URL rootUrl = new URL(root);
            URL finalUrl = new URL(rootUrl, relativePath);
            return finalUrl.toString();
        }
        catch(Exception e) {
            throw new RuntimeException(e);
        }
    }
```

Is there a better way of achieving the same result?

We believe that the OpenAPI specification supports the way that our specification files are structured and that swagger-parser is not handling relative paths correctly.

**Comment by AndySpaven** [comment]
Pete and I have raised a PR for this now.  Maybe this helps.  https://github.com/swagger-api/swagger-parser/pull/1820

**Comment by frantuma** [comment]
fixed in #1820
